R/particle_filter_state_nested.R

```diff
@@ -83,31 +84,55 @@ particle_filter_state_nested <- R6::R6Class(
       }
 
       if (!is.null(initial)) {
-        initial_data <- initial(model$info(), n_particles, pars)
-        if (is.list(initial_data)) {
-          steps <- particle_steps(steps, initial_data$step)
-          model$set_state(initial_data$state, initial_data$step)
+        initial_data <- Map(initial,
```

Comment:
Can you pull all this logic (probably everything from and including the if statement) into a free function to make it easier to contain please?